java/registry/src/main/java/dev/sunbirdrc/registry/service/impl/RegistryServiceImpl.java

```diff
@@ -150,27 +156,32 @@ public Vertex deleteEntityById(Shard shard, String userId, String uuid) throws E
                 if (!StringUtils.isEmpty(index) && index.equals(Schema)) {
                     schemaService.deleteSchemaIfExists(vertex);
                 }
-                if (!(vertex.property(Constants.STATUS_KEYWORD).isPresent()
-                        && vertex.property(Constants.STATUS_KEYWORD).value().equals(Constants.STATUS_INACTIVE))) {
-                    registryDao.deleteEntity(vertex);
+                if (isHardDeleteEnabled) {
+                    registryDao.hardDeleteEntity(vertex);
                     databaseProvider.commitTransaction(graph, tx);
                     auditService.auditDelete(
                             auditService.createAuditRecord(userId, uuid, tx, index),
                             shard);
                     if (isElasticSearchEnabled()) {
                         callESActors(null, "DELETE", index, uuid, tx);
                     }
-
-
+                } else {
+                    registryDao.deleteEntity(vertex);
+                    databaseProvider.commitTransaction(graph, tx);
+                    auditService.auditDelete(auditService.createAuditRecord(userId, uuid, tx, index),
+                            shard);
+                    if (isElasticSearchEnabled()) {
```

Comment:
this if block can be places outside of if-else statments since this is common to both